cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -1084,10 +1089,62 @@ private QueryBuilder createTermQuery(QueryTerm queryTerm, String textField, Sear
         : createTermQuery(queryTerm, NESTED_VALUE_FIELD, term));
     QueryBuilder propertyQuery = QueryBuilders.nestedQuery(PROPS_FIELD, boolQuery, ScoreMode.Max);
 
+    if (queryTerm.getSearchType() == QueryTerm.SearchType.DATE) {
+      BoolQueryBuilder dateQuery = createDateQuery(queryTerm, field, request);
+      return new BoolQueryBuilder().should(plainQuery).should(propertyQuery).should(dateQuery).minimumShouldMatch(1);
+    }
+
     // match either a plain term of the form "f:t" or the word "t" in property "f"
     return new BoolQueryBuilder().should(plainQuery).should(propertyQuery).minimumShouldMatch(1);
   }
 
+  /**
+   * Creates a bool query for a date search. If there is no field, creation times and user defined dates are queried.
+   * Otherwise, a field refers to a user defined property, which is searched for with a nested query.
+   *
+   * @param queryTerm QueryTerm object containing the term and its search properties
+   * @param field property field to search in. If undefined search all properties and creation times
+   * @param request the search request
+   * @return a structured query for a date search
+   */
+  private BoolQueryBuilder createDateQuery(QueryTerm queryTerm, @Nullable String field, SearchRequest request) {
+    if (field == null) {
+      QueryBuilder creationDateQuery = createDateRangeQuery(queryTerm, "created").boost(2);
```

Comment:
Lets use static final field `CREATED_FIELD`

Reply:
got it!